iothub/device/samples/PnpDeviceSamples/Thermostat/Program.cs

```diff
@@ -0,0 +1,104 @@
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Shared;
+using PnpHelpers;
+
+namespace TemperatureController
+{
+    public class Program
+    {
+        private const string DeviceConnectionString = "device_connection_string_here";
+        private const string ModelId = "dtmi:com:example:TemperatureController;1";
+        private const string Thermostat1 = "thermostat1";
+        private const string Thermostat2 = "thermostat2";
+
+        private static DeviceClient s_deviceClient;
+
+        public static async Task Main(string[] _)
+        {
+            await RunSampleAsync();
+        }
+
+        private static async Task RunSampleAsync()
+        {
+            // This sample follows the following workflow:
+            // -> Initialize device client instance.
+            // -> Send "current temperature" over both telemetry and property updates.
+
+            PrintLog($"Initialize the device client.");
+            await InitializeDeviceClientAsync();
+
+            PrintLog($"Send current temperature reading.");
+            await SendCurrentTemperatureAsync();
+
+            PrintLog($"Press any key to exit");
+            Console.ReadKey();
+        }
+
+        // Initialize the device client instance over Mqtt protocol, setting the ModelId into ClientOptions, and open the connection.
+        // This method also sets a connection status change callback.
+        private static async Task InitializeDeviceClientAsync()
+        {
+            var options = new ClientOptions
+            {
+                ModelId = ModelId,
+            };
+
+            // Initialize the device client instance using the device connection string, transport of Mqtt over TCP (with fallback to Websocket),
+            // and the device ModelId set in ClientOptions.
+            s_deviceClient = DeviceClient.CreateFromConnectionString(DeviceConnectionString, TransportType.Mqtt, options);
+
+            // Register a connection status change callback, that will get triggered any time the device's connection status changes.
+            s_deviceClient.SetConnectionStatusChangesHandler((status, reason) =>
+            {
+                PrintLog($"Connection status change registered - status={status}, reason={reason}");
+            });
+
+            // This will open the device client connection over Mqtt.
+            await s_deviceClient.OpenAsync();
+        }
+
+        // Send the current temperature over telemetry and reported property.
+        private static async Task SendCurrentTemperatureAsync()
+        {
+            // Send current temperature over telemetry.
+            string telemetryName = "temperature";
+
+            // Generate a random value between 40F and 90F for the current temperature reading.
+            double currentTemperature = new Random().Next(40, 90);
+
+            string telemetryPayload = $"{{ \"{telemetryName}\": {currentTemperature} }}";
```

Comment:
Can this be built using helper?

Reply:
I didn't want to create the complete Message via the helper, since that'll bring device client dependency into it, and I wanted it to be as generic as possible. 
This seemed pretty simple to build, so I left it in the sample. If we feel this is something we can add into the helper, I can create a method to simply format the telemetry payload, which I think is what you're asking for here.